test/performance/shmem_perf_suite/bi_dir.h

```diff
@@ -47,12 +44,34 @@ void static inline bi_bw_put(int len, perf_metrics_t *metric_info)
             }
             shmem_quiet();
         }
-        end = perf_shmemx_wtime();
-
-        calc_and_print_results((end - start), len, *metric_info);
+    } else {
+        for (i = 0; i < metric_info->warmup; i++) {
+            for(j = 0; j < metric_info->window_size; j++) {
+#ifdef USE_NONBLOCKING_API
+                shmem_putmem_nbi(metric_info->dest, metric_info->src, len, dest);
+#else
+                shmem_putmem(metric_info->dest, metric_info->src, len, dest);
+#endif
+            }
+            shmem_quiet();
+        }
+    }
```

Comment:
Are the if and else cases identical?  If so ... why have the conditional?

Reply:
oh, yes. Sorry that I missed this. Earlier the if and else cases was different because of the timers. But now since we are handling this outside this scope, we can code this better. This change will only impact the bi-dir case.

Reply:
I agree for the warm-up the cases are doing the same thing, and this is also true for the second section where we are doing timing and all we really need is 
 if (streaming_node(*metric_info)) {
        start = perf_shmemx_wtime();